pkg/utils/pulsar_util.go

```diff
@@ -0,0 +1,85 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package utils
+
+import (
+	"context"
+	"github.com/apache/pulsar-client-go/pulsar"
+	"github.com/sirupsen/logrus"
+	"time"
+)
+
+var pulsarClient pulsar.Client
+
+func init() {
+	var err error
+	pulsarClient, err = pulsar.NewClient(pulsar.ClientOptions{URL: "pulsar://localhost:6650"})
+	if err != nil {
+		logrus.Error("init pulsar client failed.")
+	}
+}
+
+func ReceiveEarliestMsg(topic string) pulsar.Message {
+	readMsg := make(chan pulsar.ReaderMessage)
+	readerOptions := pulsar.ReaderOptions{
+		Topic:          topic,
+		Name:           "OFFSET_LIST" + topic,
+		StartMessageID: pulsar.EarliestMessageID(),
+		MessageChannel: readMsg,
+	}
+	return receiveMsgByMsgId(readerOptions)
+}
+
+func ReceiveLastedMsg(topic string) pulsar.Message {
+	readMsg := make(chan pulsar.ReaderMessage)
+	readerOptions := pulsar.ReaderOptions{
+		Topic:          topic,
+		Name:           "OFFSET_LIST" + topic,
```

Comment:
duplicate hard code of "OFFSET_LIST", it's fine to name `OFFSET_LIST` without the topic